### PR TITLE
Internal: clean up initial state creation

### PR DESCRIPTION
**Pull request by odesenfans**
Problem: the initial state creation for tests is messy and hard to
maintain.

Solution: introduce a builder pattern to make it easier to add contracts
and configure the initial state.

Issue Number: N/A

## Type

- [ ] feature
- [ ] bugfix
- [x] dev (no functional changes, no API changes)
- [ ] fmt (formatting, renaming)
- [ ] build
- [ ] docs
- [ ] testing

## Breaking changes?

- [ ] yes
- [x] no

### CODE REVIEW

**Comment by odesenfans** [comment]
Depends on #215.

File: tests/integration/common/state.rs [hunk 1]
```diff
@@ -85,25 +98,286 @@ fn init_logging() {
...
+#[derive(Debug)]
+struct FeeConfig {
+    strk_fee_token_address: ContractAddress,
+    eth_fee_token_address: ContractAddress,
+    initial_balance: u128,
+}
+
+/// Builds the initial state for OS integration tests.
+///
+/// This builder allows to deploy contracts and configure account balances and then generates
```
Comment (notlesh) [inline tests/integration/common/state.rs:125]: Did this comment get cut off?

File: tests/integration/common/state.rs [hunk 2]
```diff
@@ -85,25 +98,286 @@ fn init_logging() {
...
+    cairo1_contracts: HashMap<String, Cairo1Contract>,
+    /// Config for fees: initial balance, fee token contract addresses, etc.
+    fee_config: Option<FeeConfig>,
+    block_context: &'a BlockContext,
+    /// Contract address generator.
+    address_generator: StdRng,
+}
+
+impl<'a> StarknetStateBuilder<'a> {
+    fn new(block_context: &'a BlockContext) -> Self {
```
Comment (notlesh) [inline tests/integration/common/state.rs:140]: Any reason not to make these `pub`?

**Review by notlesh — APPROVED** [review]
The builder pattern is a great idea here :100:

File: tests/integration/common/state.rs (hunk 2 above)
Comment (odesenfans) [reply tests/integration/common/state.rs:140]: No, it's true that there's a lot of private methods so I'll make the ones that need to be `pub`.

**Pull request merged**